src/shadowsocks_config.ts

```diff
@@ -338,3 +339,63 @@ export const SIP002_URI = {
     return `ss://${userInfo}@${uriHost}:${port.data}/${queryString}${hash}`;
   },
 };
+
+export interface OnlineConfig {
+  // URL endpoint to retrieve a Shadowsocks configuration.
+  readonly url: string;
+  // Any additional configuration (e.g. `certFp`, `httpMethod`, etc.) may be stored here.
+  readonly extra: {[key: string]: string};
+}
+
+// Ref: https://github.com/shadowsocks/shadowsocks-org/issues/89
+export const SIP008_URI = {
+  PROTOCOL: 'ssconf',
+
+  validateProtocol: (uri: string) => {
+    if (!uri || !uri.startsWith(SIP008_URI.PROTOCOL)) {
+      throw new InvalidUri(`URI must start with "${SIP008_URI.PROTOCOL}"`);
+    }
+  },
+
+  parse: (uri: string): OnlineConfig => {
+    SIP008_URI.validateProtocol(uri);
+
+    // URL parser for expedience, replacing the protocol "ssconf" with "https" to ensure correct
+    // results, otherwise browsers like Safari fail to parse it.
+    const inputForUrlParser = uri.replace(new RegExp(`^${SIP008_URI.PROTOCOL}`), 'https');
+    // The built-in URL parser throws as desired when given URIs with invalid syntax.
+    const urlParserResult = new URL(inputForUrlParser);
+    // Use ValidatedConfigFields subclasses (Host, Port, Tag) to throw on validation failure.
+    const uriFormattedHost = urlParserResult.hostname;
+    let host: Host;
+    try {
+      host = new Host(uriFormattedHost);
+    } catch (_) {
+      // Could be IPv6 host formatted with surrounding brackets, so try stripping first and last
```

Comment:
I believe an IPv6 host will * always* have the brackets, since that's in the URL standard: https://tools.ietf.org/html/rfc3986#section-3.2.2

Reply:
Let's update the comment:
```
// The IPv6 host has surrounding brackets, so strip them.
```

Reply:
I think the comment is acurate, because it __could__ be an IPv6 host, but also an invalid IPv4 address or domain.